composer/callbacks/checkpoint_saver.py

```diff
@@ -297,29 +271,44 @@ def __init__(
         self.save_interval = save_interval
         self.last_checkpoint_batch: Optional[Time] = None
 
-        self.folder = folder
-
-        self.filename = PartialFilePath(filename.lstrip('/'), folder)
-        self.latest_filename = PartialFilePath(latest_filename.lstrip('/'), folder) if latest_filename else None
-
-        self.artifact_name = PartialFilePath(artifact_name) if artifact_name else None
-        self.latest_artifact_name = PartialFilePath(latest_artifact_name) if latest_artifact_name else None
+        
+        self.folder = urlparse(folder).path
+        backend = urlparse(folder).scheme
+        self.backend = (backend if backend != '' else None)
+        self.folder = self.folder.lstrip('/')
+        
 
+        self.filename = filename
+        self.latest_filename = latest_filename
         self.overwrite = overwrite
         self.saved_checkpoints: List[str] = []
         self.num_checkpoints_to_keep = num_checkpoints_to_keep
         self.weights_only = weights_only
+        self.checkpoint_file_path = str(Path(self.folder) / Path(self.filename))
+        if latest_filename:
```

Comment:
prefer being explicit about which version of truthiness your relying on here. so `if latest_filename is None or len(latest_filename) == 0:`

Reply:
fixed